Framework/API/test/WorkspaceListPropertyTest.h

```diff
@@ -0,0 +1,304 @@
+  // PropertyManager interfaces (such as Algorithm).
+  //------------------------------------------------------------------------------
+  void test_setPropertyValue_throws_with_workspace_not_in_ADS() {
+    MyAlgorithm alg;
+    alg.initialize();
+    TSM_ASSERT_THROWS("Workspaces are not in the ADS, so should throw",
+                      alg.setPropertyValue("MyProperty", "a"),
+                      std::invalid_argument &);
+  }
+
+  void test_setPropertyValue_with_single_workspace() {
+    AnalysisDataServiceImpl &ads = AnalysisDataService::Instance();
+    ads.add("a", boost::make_shared<WorkspaceTester>());
+
+    MyAlgorithm alg;
+    alg.initialize();
+    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("MyProperty", "a"));
+
+    ads.remove("a");
+  }
+
+  void test_septPropertyValue_with_multiple_workspaces() {
+    AnalysisDataServiceImpl &ads = AnalysisDataService::Instance();
+    ads.add("a", boost::make_shared<WorkspaceTester>());
+    ads.add("b", boost::make_shared<WorkspaceTester>());
+
+    MyAlgorithm alg;
+    alg.initialize();
+    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("MyProperty", "a, b"));
+
+    ads.remove("a");
+    ads.remove("b");
+  }
+
+  void test_septPropertyValue_with_multiple_workspaces_last_doesnt_exist() {
+    AnalysisDataServiceImpl &ads = AnalysisDataService::Instance();
+    ads.add("a", boost::make_shared<WorkspaceTester>());
+
+    MyAlgorithm alg;
+    alg.initialize();
+    TSM_ASSERT_THROWS("Not ALL Workspaces are in the ADS, so should throw",
+                      alg.setPropertyValue("MyProperty", "a, b"),
+                      std::invalid_argument &);
+
+    ads.remove("a");
+  }
+
+  void test_set_and_get_Property_as_vector() {
+    auto a = boost::make_shared<WorkspaceTester>();
+    auto b = boost::make_shared<WorkspaceTester>();
+
+    MyAlgorithm alg;
+    alg.initialize();
+
+    auto vecIn = std::vector<Workspace_sptr>();
+    vecIn.push_back(a);
+    vecIn.push_back(b);
+
+    alg.setProperty("MyProperty", vecIn);
+    // Now fetch the property.
+    std::vector<boost::shared_ptr<DataItem>> vecOut = alg.getProperty("MyProperty");
+
+    TS_ASSERT_EQUALS(vecIn.size(), vecOut.size());
+
+	auto testVec = std::vector<Workspace_sptr>();
+
+	for (auto &i : vecOut) {
+		testVec.push_back(boost::dynamic_pointer_cast<WorkspaceTester>(i));
+	}
+
+    TS_ASSERT_EQUALS(vecIn, testVec);
+  }
+
+  void test_getPropertyValue() {
+    AnalysisDataServiceImpl &ads = AnalysisDataService::Instance();
+    auto a = boost::make_shared<WorkspaceTester>();
+    auto b = boost::make_shared<WorkspaceTester>();
+    ads.add("a", a);
+    ads.add("b", b);
+
+    MyAlgorithm alg;
+    alg.initialize();
+
+    auto vecIn = std::vector<Workspace_sptr>();
+    vecIn.push_back(a);
+    vecIn.push_back(b);
+
+    alg.setProperty("MyProperty", vecIn);
+    //Now fetch the property.
+    std::string outStr = alg.getPropertyValue("MyProperty");
```

Comment:
I'm really not sure what we ought to use to represent the workspace "value". Name would make sense if all workspaces resided in the ADS, but is `value == std::string::empty()` the correct measure when the workspace has no name?